File: src/main/java/com/catworldspace/plugin/system/event/PostLoginEventHandler.java
package com.catworldspace.plugin.system.event;

import com.catworldspace.plugin.system.common.CommandHelper;
import net.md_5.bungee.api.ChatColor;
import net.md_5.bungee.api.ProxyServer;
import net.md_5.bungee.api.config.ServerInfo;
import net.md_5.bungee.api.connection.ProxiedPlayer;
import net.md_5.bungee.api.event.PostLoginEvent;
import net.md_5.bungee.api.plugin.Listener;
import net.md_5.bungee.event.EventHandler;

import java.util.UUID;
import java.util.logging.Logger;

public class PostLoginEventHandler implements Listener {

    // TODO: 读取自 config.yml 的值

    private final boolean IsDebug = true;
    private final String namePrefixDivider = ".";
    private final String uuidPrefix = "00000000-0000-0000-";
    private final String server1Name = "LoginServer";
    private final String server2Name = "LoginServer_be";

    @EventHandler
    public void onPostLogin(PostLoginEvent event) {
        Logger logger = ProxyServer.getInstance().getLogger();

        ProxiedPlayer player = event.getPlayer();

        UUID uuid = player.getUniqueId();
        String name = player.getName();

        // TODO：动态配置, 将文本移至本地化
        int prefixIndex = name.indexOf(namePrefixDivider);
        if (prefixIndex == -1) { // 未找到前缀分隔符
            if (IsDebug) {
                logger.info(CommandHelper.GetStringWithPrefix("玩家 " + player.getName() + " 未被分配前缀"));
            }
        } else {
            if (!uuid.toString().startsWith(uuidPrefix)) { // UUID 未以指定字符串开头
                if (IsDebug) {
                    logger.info(CommandHelper.GetStringWithPrefix("玩家 " + player.getName() + " UUID验证失败"));
                }
            } else {
                String prefix = name.substring(0, prefixIndex);
                //if (IsDebug) {
                //    logger.info(CommandHelper.GetTextWithPrefix("玩家 "+ player.getName() + " 的前缀为 " + prefix));
                //}

                ServerInfo server;
                // TODO: 使用配置中的设置
                if (prefix == server2Name) {
                    server = CommandHelper.GetServerByName(server2Name);
                    return;
                } else {
                    server = CommandHelper.GetServerByName(prefix);
                }

                if (server == null) {
                    if (IsDebug) {
                        logger.warning("未找到与前缀 " + prefix + " 匹配的服务器");
                    }
                    player.disconnect(CommandHelper.CreateMessage(ChatColor.RED + "未找到匹配的服务器."));
                } else {
                    CommandHelper.ConnectPlayerTo(player,server);
                }
            }
        }

        if (IsDebug) {
            CommandHelper.ConnectPlayerTo(player,CommandHelper.GetServerByName(server1Name));
        } else {
            CommandHelper.DisconnectPlayer(player,CommandHelper.CreateMessage(ChatColor.RED + "你不在服务器白名单上!"));
        }
    }


}

File: src/main/java/com/catworldspace/plugin/system/events/PostLoginEventHandler.kt
package com.catworldspace.plugin.system.events

import com.catworldspace.plugin.system.common.CommandHelper.createPrefixedMessage
import com.catworldspace.plugin.system.models.redirect.RedirectResultEnum
import com.catworldspace.plugin.system.common.RedirectManager.tryLoginIn
import com.catworldspace.plugin.system.common.connection.ServerHelper.getServerByName
import com.catworldspace.plugin.system.common.program.Strings
import net.md_5.bungee.api.ChatColor
import net.md_5.bungee.api.ProxyServer
import net.md_5.bungee.api.event.PostLoginEvent
import net.md_5.bungee.api.plugin.Listener
import net.md_5.bungee.event.EventHandler

class PostLoginEventHandler : Listener {
    // TODO: 读取自 config.yml 的值
    private val isDebug = true

    private val serverName1 = "LoginServer"
    private val serverName2 = "LoginServer_be"


    @EventHandler
    fun onPostLogin(event: PostLoginEvent) {
        val logger = ProxyServer.getInstance().logger

        val player = event.player
        val result = tryLoginIn(player)

        when (result.resultEnum) {
            RedirectResultEnum.PLAYER_NO_PREFIX -> logger.info(
                "${Strings.consolePrefix} 玩家 ${player.name} 未被分配前缀"
            )

            RedirectResultEnum.PLAYER_UUID_VERIFY_FAIL -> logger.info(
                "${Strings.consolePrefix} 玩家 ${player.name} UUID 验证失败"
            )

            RedirectResultEnum.NO_PREFIX_MATCHED_SERVER -> logger.warning(
                "${Strings.consolePrefix} 未找到与前缀匹配的服务器"
            )

            RedirectResultEnum.OK -> {
                if (result.server != null) {
                    player.connect(result.server)
                    logger.info(
                        "${Strings.consolePrefix} 已将玩家 ${player.name} 连接至 ${result.server.name} 服务器"
                    )
                } else {
                    logger.warning(
                        "${Strings.consolePrefix} 未预见的错误"
                    )
                }
            }
        }

        if (isDebug) {
            player.connect(getServerByName(serverName1))
        }
        player.disconnect(*createPrefixedMessage("${ChatColor.RED} 未找到匹配的服务器, 请联系管理员"))
        logger.info(
            "${Strings.consolePrefix} 已拒绝玩家 ${player.name} 的连接请求"
        )
    }
}